### PR TITLE
Pract 24/fixed authentication branch

### PR DESCRIPTION
**Pull request by Jukari-Richardson**
_For an example of how to fill this template out, [see this Pull Request](https://github.com/the-collab-lab/tcl-3-smart-shopping-list/pull/44)._

## Description
I am pushing this file after correcting  multiple merge issues with the main branch and errors regarding user authentication which were not operating properly during Postman testing. Initially the server was calling unnecessary or misplaced middleware before allowing the routes to be called first. After addressing this it seems to be using our authentication functions properly enacting the cookies and jwt token.

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|     | :bug: Bug fix              |
| ✓   | :sparkles: New feature     |
|     | :hammer: Refactoring       |
|     | :100: Add tests            |
|     | :link: Update dependencies |
|     | :scroll: Docs              |

## Testing Steps / QA Criteria
User should receive a jwt token and A user id during registering for profile which will then be used to login to their profile 

© 2022 GitHub, Inc.
Terms
Privacy
Security
Status

### CODE REVIEW

File: src/middleware/error-handler.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  let customError = {
+    // set default
+    statusCode: err.statusCode || StatusCodes.INTERNAL_SERVER_ERROR,
+    msg: err.message || 'Something went wrong try again later',
+  }
+
+  // if (err instanceof CustomAPIError) {
+  //   return res.status(err.statusCode).json({ msg: err.message })
+  // }
+
```
Comment (koral14) [inline src/middleware/error-handler.js:13]: please remove the commented out code

**Review by koral14 — APPROVED** [review]
Well done. Please just remove the commented out code from error-jandler.js

File: middleware/authentication.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
+const User = require('../models/User')
```
Comment (koral14) [inline middleware/authentication.js:1]: you are not using the User variable, so you an remove it

**Comment by ddigioia3** [comment]
It looks like you created a whole seperate auth workflow in seperate folders rather than updating the existing.
This isn't following best practices / existing norms in our project for how the folders should be setup, but worse yet its almost certainly got bugs that I couldn't easily find in a review because I can't really side by side the old and new.

**Review by madebymiguel — APPROVED** [review]
LGMT

**Review by nikkigraybeal — APPROVED** [review]

**Pull request merged**